app/javascript/components/Header/index.js

```diff
@@ -1,40 +1,237 @@
+  HeaderItem,
+  NextItem,
+} from '@zendeskgarden/react-dropdowns'
+import { Avatar } from '@zendeskgarden/react-avatars'
+import TranslationIcon from '@zendeskgarden/svg-icons/src/12/translation-exists-fill.svg'
+import { MD } from '@zendeskgarden/react-typography'
 
-import ContentIcon from 'material-ui/svg-icons/content/create'
 import UpdateUserOfficeMutation from '/mutations/UpdateUserOfficeMutation.gql'
 import { UserContext, FilterContext } from '/context'
-import { present } from '../../lib/utils'
-import s from './main.css'
 
-const styles = {
-  item: {
-    fontSize: 14,
-  },
+const NavigationText = styled.div`
+  color: #08c;
+`
+
+const UserDetails = styled.div`
+  margin-left: 10px;
+`
+
+const UserProfileContainer = styled.button`
+  display: flex;
+  border: none;
+  text-align: left;
+`
+
+const UserName = styled(MD)`
+  font-weight: bold;
+`
+
+const UserProfile = ({ currentUser, offices, handleOfficeSelect, location, router }) => {
+  const { i18n, t } = useTranslation()
+  const languages = [
+    { label: 'English', value: 'en' },
+    // Enable when Japanense is supported
+    // { label: '日本語', value: 'ja' }
+  ]
+
+  const [isOpen, setIsOpen] = useState(false)
+  const [tempSelectedItem, setTempSelectedItem] = useState()
+  const [selectedItem, setSelectedItem] = useState({ office: currentUser.office, language: languages[0] })
+
+  const handleStateChange = (changes, stateAndHelpers) => {
+    if (R.hasPath(['isOpen'])(changes)) {
+      setIsOpen(
+        changes.selectedItem === 'default-office' ||
+          changes.selectedItem === 'language-settings' ||
+          changes.selectedItem === 'general-settings' ||
+          changes.isOpen
+      )
+    }
+
+    if (R.hasPath(['selectedItem'])(changes)) {
+      const itemSelected = R.prop('selectedItem')(changes)
+      if (itemSelected === 'general-settings') {
+        switch (tempSelectedItem) {
+          case 'default-office':
+            stateAndHelpers.setHighlightedIndex(0)
+            break
+          case 'language-settings':
+            stateAndHelpers.setHighlightedIndex(1)
+        }
+      }
+      setTempSelectedItem(itemSelected)
+    }
+  }
+
+  const handleSelect = selectedItem => {
+    if (tempSelectedItem === 'default-office') {
+      if (R.hasPath(['office'])(selectedItem)) {
+        handleOfficeSelect(selectedItem.office)
+        setSelectedItem(selectedItem)
+      }
+    } else if (tempSelectedItem === 'language-settings') {
+      if (R.hasPath(['language'])(selectedItem)) {
+        i18n.changeLanguage(selectedItem.language.value, () => {
+          // TODO: Handle callback (error/success)
+        })
+        setSelectedItem(selectedItem)
+      }
+    } else if (selectedItem === 'admin') {
+      router.push('/portal/admin')
+    } else if (selectedItem === 'home') {
+      router.push('/portal')
+    } else if (selectedItem === 'sign-out') {
+      router.push('/portal/sign_out')
+    }
+  }
+
+  const renderItems = () => {
```

Comment:
There is quite a lot going on in this function. Separating sub components out into their own files under the same folder can help readability and separation of concerns.
i.e. The UserProfile component could just care about what to show and not also how to show each thing in the menu.